examples/evolve_dstream_surf.py

```diff
@@ -47,9 +59,16 @@ def evolve_drift(population: typing.List) -> typing.List:
             parent.CreateOffspring()
             for parent in selector.choices(population[:nsplit], k=nsplit)
         ]
+        if fossil_interval and generation % fossil_interval == 0:
+            fossils.extend(
+                [
+                    parent.CreateOffspring() for parent in
+                    selector.sample(population, k=int(len(population) * fossil_sample_percentage))
+                ]
+            )
         selector.shuffle(population)
 
-    return population
+    return fossils + population
```

Comment:
a pattern I like that makes the return type obvious at a glance
```suggestion
    return [*fossils, *population]
```